### PR TITLE
chore: add check for booking

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

Everything works fine in handleInstantMeeting and it shouldn't display any toast.

https://calcominc.slack.com/archives/C04CWEU35NV/p1720712414218979

BEFORE:-

Screenshot 2024-07-11 at 9 30 17 PM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A  I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1) Create an org team event type
2) Enable instant meeting
3) go to booking page and click on bottom modal "connect now" and fill the form

You shouldn't see any toast error

### CODE REVIEW

File: packages/features/bookings/Booker/components/hooks/useBookings.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const useBookings = ({ event, hashedLink, bookingForm, metadata, teamMemb
     function refactorMeWithoutEffect() {
       const data = _instantBooking.data;
 
-      if (!data) return;
+      if (!data || !data.booking) return;
```
Comment (Udit-takkar) [inline packages/features/bookings/Booker/components/hooks/useBookings.ts:96]: data.booking is not present when host hasn't accepted the meeting

File: packages/trpc/server/routers/publicViewer/noShow.handler.ts [hunk 2]
```diff
@@ -7,9 +7,9 @@ type NoShowOptions = {
 };
 
 export const noShowHandler = async ({ input }: NoShowOptions) => {
-  const { bookingUid, attendees } = input;
+  const { bookingUid, attendees, noShowHost } = input;
 
-  return handleMarkNoShow({ bookingUid, attendees });
+  return handleMarkNoShow({ bookingUid, attendees, noShowHost });
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/publicViewer/noShow.handler.ts:12]: not related to PR but i found this was missing while working on https://github.com/calcom/cal.com/pull/15738

File: packages/features/bookings/Booker/components/hooks/useBookings.ts (hunk 1 above)
Comment (joeauyeung) [reply packages/features/bookings/Booker/components/hooks/useBookings.ts:96]: I'm looking at the return of `getInstantBookingLocation` and the only thing it's returning is `booking`. Do we even need this line? Would there be any side effects if we just deleted it?

Comment (Udit-takkar) [reply packages/features/bookings/Booker/components/hooks/useBookings.ts:96]: yeah we need it to get the daily video call url from it and  set the InstantVideoMeetingUrl

Comment (Udit-takkar) [reply packages/features/bookings/Booker/components/hooks/useBookings.ts:96]: booking is set to ACCEPTED and returned only when host accepts it.

**Review by CarinaWolli — APPROVED** [review]
Code looks good!

**Pull request merged**